File: server/models/AdminMember.js
const mongoose = require('mongoose');

const adminMemberSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  username: {
    type: String,
    required: true,
    unique: true,
    lowercase: true,
    trim: true
  },
  password: {
    type: String,
    required: true
  },
  role: {
    type: String,
    enum: ['owner', 'editor'],
    default: 'editor'
  },
  createdBy: {
    type: String,
    default: 'Owner'
  }
}, { timestamps: true });

module.exports = mongoose.model('AdminMember', adminMemberSchema);

File: server-java/src/main/java/com/denesens/model/AdminMember.java
package com.denesens.model;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Document(collection = "adminmembers")
public class AdminMember {
    @Id
    private String id;
    
    private String name;

    @Indexed(unique = true)
    private String username;
    
    private String password;
    
    private String role = "editor";
    
    private String createdBy = "Owner";

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}
